TST: SDSS remote data not found (404), failing 6 tests

Example log: https://github.com/astropy/specutils/actions/runs/8994189000/job/24707287554

```
FAILED ../../specutils/tests/test_loaders.py::test_sdss_spspec - urllib.error.HTTPError: HTTP Error 404: Not Found
FAILED ../../specutils/tests/test_loaders.py::test_sdss_spspec_stream - urllib.error.HTTPError: HTTP Error 404: Not Found
FAILED ../../specutils/tests/test_loaders.py::test_sdss_compressed[gzip] - urllib.error.HTTPError: HTTP Error 404: Not Found
FAILED ../../specutils/tests/test_loaders.py::test_sdss_compressed[bzip2] - urllib.error.HTTPError: HTTP Error 404: Not Found
FAILED ../../specutils/tests/test_loaders.py::test_sdss_compressed[xz] - urllib.error.HTTPError: HTTP Error 404: Not Found
FAILED ../../specutils/tests/test_loaders.py::test_sdss_spplate - urllib.error.HTTPError: HTTP Error 404: Not Found
```

1. Ask SDSS if this is temporary.
2. If not, find replacement data or create fake local data to mimic the desired format.

--- a/specutils/tests/test_loaders.py
+++ b/specutils/tests/test_loaders.py
@@ -202,7 +202,7 @@ def test_sdss_spec(tmp_path):
 @pytest.mark.remote_data
 def test_sdss_spspec(tmp_path):
     sp_pattern = 'spSpec-51957-0273-016.fit'
-    url = f'http://das.sdss.org/spectro/1d_26/0273/1d/{sp_pattern}'
+    url = f'https://das.sdss.org/spectro/1d_26/0273/1d/{sp_pattern}'
     with urllib.request.urlopen(url) as response:
         # Read from open file object
         spec = Spectrum1D.read(response, format="SDSS-I/II spSpec")
@@ -256,7 +256,7 @@ def test_sdss_spspec_stream():
     """Test direct read and recognition of SDSS-I/II spSpec from remote URL,
     i.e. do not rely on filename pattern.
     """
-    sdss_url = 'http://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit'
+    sdss_url = 'https://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit'
     spec = Spectrum1D.read(sdss_url)
 
     assert isinstance(spec, Spectrum1D)
@@ -279,7 +279,7 @@ def test_sdss_compressed(compress, tmp_path):
 
     # Deliberately not using standard filename pattern to test header info.
     tmp_filename = tmp_path / 'SDSS-I.fits'
-    with urllib.request.urlopen('http://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit') as response:
+    with urllib.request.urlopen('https://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit') as response:
         with open(tmp_filename, 'wb') as tmp_file:
             shutil.copyfileobj(response, tmp_file)
 
@@ -307,7 +307,7 @@ def test_sdss_compressed(compress, tmp_path):
 def test_sdss_spplate(tmp_path):
     """Test loading of multi-object spectrum from SDSS `spPlate` format FITS file.
     """
-    with urllib.request.urlopen('http://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit') as response:
+    with urllib.request.urlopen('https://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit') as response:
         # Read reference spectrum from open file object
         with warnings.catch_warnings():
             warnings.simplefilter('ignore', FITSFixedWarning)

